templates/console/source/lambda/config-version/cf_config_version_exporter/cf_config_version_exporter.py

```diff
@@ -12,33 +12,36 @@
 DDB_LATESTVERSION_TABLE_NAME = os.environ['DDB_LATESTVERSION_TABLE_NAME']
 DDB_SNAPSHOT_TABLE_NAME = os.environ['DDB_SNAPSHOT_TABLE_NAME']
 
-# TODO: for local debug purpose, will remove before release
-# S3_BUCKET = "cloudfrontconfigversions-cloudfrontconfigversions-60jwdz7zg1zi"
-# DDB_VERSION_TABLE_NAME = 'CloudFrontConfigVersionStack-CloudFrontConfigVersionTable6E23F7F5-D8I07GGNBYFJ'
-# DDB_LATESTVERSION_TABLE_NAME = 'CloudFrontConfigVersionStack-CloudFrontConfigLatestVersionTable44770AF8-7LF5V48RKDK0'
 
 log = logging.getLogger()
 log.setLevel('INFO')
 
 
 def lambda_handler(event, context):
     # extract the distribution id from the input
+    distribution_id = get_distributionId(event)
+
+    cf_client = boto3.client('cloudfront')
+    s3_client = boto3.client('s3')
+    ddb_client = boto3.resource('dynamodb')
+    return update_config_version(distribution_id, cf_client, s3_client, ddb_client)
+
+
+def get_distributionId(event):
     log.info(event['detail'])
-    if event['detail']['eventName'] == 'CreateDistribution' or event['detail']['eventName'] == 'CreateDistributionWithTags':
+    if event['detail']['eventName'] == 'CreateDistribution' or event['detail'][
+        'eventName'] == 'CreateDistributionWithTags':
         response_parameters = event["detail"]["responseElements"]
         distribution_id = response_parameters['distribution']['id']
     else:
         request_parameters = event["detail"]["requestParameters"]
         distribution_id = request_parameters["id"]
-
     log.info(distribution_id)
+    return distribution_id
 
-    return update_config_version(distribution_id)
 
-
-def update_config_version(distribution_id):
+def update_config_version(distribution_id, cf_client, s3_client, ddb_client):
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This function contains 45 lines of code, not including blank lines or lines with only comments, Python punctuation characters, identifiers, or literals. By comparison, 98% of the functions in the CodeGuru reference dataset contain fewer lines of code. Large functions might be difficult to read and have logic that is hard to understand and test.

We recommend that you simplify this function or break it into multiple functions. For example, consider extracting the code block on lines 65-155 into a separate function.